feat(apple): add troubleshooting for Cocoapods user script sandbox setup (#12742)
getsentry/sentry-docs: docs/platforms/apple/common/install/cocoapods.mdx

```

Afterwards run `pod install`.

> In case you are using Xcode 14 or later, you might have to disable the "Enable User Script Sandbox" option in the target settings by setting it to `NO`. See [Troubleshooting](/platforms/apple/common/troubleshooting) for more information.

Reviewer: `h`: You need to use `<PlatformLink to= ...` Just search for it in the codebase and you will see how to use it.

> In case you are using Xcode 14 or later, you might have to disable the "Enable User Script Sandbox" option in the target settings by setting it to `NO`. See <PlatformLink to="/troubleshooting#sandbox-error-using-cocoapods">Troubleshooting</PlatformLink> for more information.

Reviewer: `m`: I think we should briefly mention the error here as well cause devs tend to overlook this.
```
error: Sandbox: rsync.samba(42924) 
```

Author: ```suggestion
In case you are using Xcode 14 or later, your project might fail to build due to `error: Sandbox: rsync.samba(42924)`. To resolve this you have to disable the "Enable User Script Sandbox" option in the target settings by setting it to `NO`. See <PlatformLink to="/troubleshooting#sandbox-error-using-cocoapods">Troubleshooting</PlatformLink> for more information.
```

> In case you are using Xcode 14 or later, your project might fail to build due to `error: Sandbox: rsync.samba(42924)`. To resolve this you have to disable the "Enable User Script Sandbox" option in the target settings by setting it to `NO`. See <PlatformLink to="/troubleshooting#sandbox-error-using-cocoapods">Troubleshooting</PlatformLink> for more information.

Reviewer 2: ```suggestion
If you're using Xcode 14 or later, your project might fail to build due to `error: Sandbox: rsync.samba(42924)`. To resolve this you have to disable the "Enable User Script Sandbox" option in the target settings by setting it to `NO`. See <PlatformLink to="/troubleshooting#sandbox-error-using-cocoapods">Troubleshooting</PlatformLink> for more information.
```
